test/PHPUnit/Controller/AbstractControllerTestCaseTest.php

```diff
@@ -538,6 +539,52 @@ public function testAssertNotTemplateName(): void
         $this->assertNotTemplateName('template/does/not/exist');
     }
 
+    /**
+     * Test case for a controller returning a view with deeply nested children
+     * View hierarchy:
+     *   layout/layout -> baz/index/childview -> child1 -> child3
+     *                                        -> child2
+     */
+    public function testSearchTemplatesVerifiesDeeplyNestedTemplateName(): void
+    {
+        $this->dispatch('/childview');
+
+        // Check that the rendered content
+        $this->assertQueryContentContains('p', 'Parent');
+        $this->assertQueryContentContains('p', 'Child 1');
+        $this->assertQueryContentContains('p', 'Child 2');
+        $this->assertQueryContentContains('p', 'Child 3');
+
+        $this->assertTemplateName('layout/layout');
+        $this->assertTemplateName('baz/index/childview');
+        $this->assertTemplateName('child1');
+        $this->assertTemplateName('child2');
+        $this->assertTemplateName('child3');
+        $this->assertNotTemplateName('foo');
+
+        // Check that the test fails when template is NOT found where it was supposed to found
+        try {
+            $this->assertTemplateName('foo');
+            $receivedException = false;
+        } catch (Exception $exception) {
+            $receivedException = true;
+        }
+        if (! $receivedException) {
+            $this->fail('Expected Exception not thrown');
+        }
+
+        // Check when template is found where it was NOT supposed to found
+        try {
+            $this->assertNotTemplateName('child1');
+            $receivedException = false;
+        } catch (Exception $exception) {
+            $receivedException = true;
+        }
+        if (! $receivedException) {
+            $this->fail('Expected Exception not thrown');
+        }
+    }
```

Comment:
Those are separate test cases so they should be split into separate tests where purpose is stated in the test name rather than a comment

Reply:
Tough crowd... 
Done